File: oldData/Python/Pro/약수의 합.py
# 약수의 합
# 문제 설명
# 정수 n을 입력받아 n의 약수를 모두 더한 값을 리턴하는 함수, solution을 완성해주세요.

# 제한 사항
# n은 0 이상 3000이하인 정수입니다.

def solution(n):
    answer = 0
    for i in range(n+1):
        if i == 0:
            continue
        if n % i == 0:
            answer += i
    return answer

# 약수는 해당값으로 나눴을떄 0으로 떨어진다
# 0부터 n까지 1씩 올리면서 딱 나누어 떨어지면 결과값에 더해서 출력한다

File: JavaScript/test/약수의 합.js
function solution(n) {
    let answer = 0;
    // 주어진 n값의 값을 i에 넣는다. i에 값은 반복할때마다 1씩 감소한다. i 값이 0이되면 실행중단.
    for(let i = n; i > 0; i--){
        // 나머지연산으로 값이 떨어지면 answer의 나눠지는 약수의 값을 더해준다.
        if((n%i) == 0){
            answer += i
        }
    }
    return answer;
}